Sami/Parser/NodeVisitor.php

```diff
@@ -151,8 +152,12 @@ protected function addMethod(ClassMethodNode $node)
             if ($param->default) {
                 $parameter->setDefault($this->context->getPrettyPrinter()->prettyPrintExpr($param->default));
             }
-            if ((string) $param->type) {
-                $parameter->setHint($this->resolveHint(array(array((string) $param->type, false))));
+            if ($type = (string) $param->type) {
+                if( $param->type instanceof FullyQualified && strpos($type, '\\') !== 0 ) {
```

Comment:
You are right !

When the parameter is typed as `array`, the `$param->type` property is a string. I've checked with PHP7 and with the types `string` or `float` for the moment there is a problem (it return a `FullyQualified` to `A\B\string`. Maybe another PR for that :smile:.